toolchain/check/context.h

```diff
@@ -222,6 +222,17 @@ class Context {
       std::optional<llvm::function_ref<auto()->DiagnosticBuilder>> diagnoser =
           std::nullopt) -> bool;
 
+  // Returns the type `type_id` as a complete type, or produces an incomplete
+  // type error and returns an error type. This is a convenience wrapper around
+  // TryToCompleteType.
+  auto AsCompleteType(
+      SemIR::TypeId type_id,
+      std::optional<llvm::function_ref<auto()->DiagnosticBuilder>> diagnoser =
+          std::nullopt) -> SemIR::TypeId {
```

Comment:
```suggestion
      std::optional<llvm::function_ref<auto()->DiagnosticBuilder>> diagnoser) -> SemIR::TypeId {
```

All the callers pass a diagnoser, maybe it should be required?